File: src/abc284/c.py

def dfs(c):
    vis[c-1] = True
    for i in range(1,n+1):
        if (not vis[i-1]) & g[c-1][i-1]:
            dfs(i)
    return 0

# main
n, m = list(map(int, input().split()))
g0 = [list(map(int, input().split())) for _ in range(m)]

g = list(list(False for _ in range(n)) for _ in range(n))
for (u,v) in g0:
    g[u-1][v-1] = True
    g[v-1][u-1] = True

vis = list(False for _ in range(n))

ans = 0
for i in range(1,n+1):
    if not vis[i-1]:
        ans += 1
        dfs(i)
print(ans)

File: src/abc284/c.rs
use proconio::input;

static mut G: Vec<Vec<usize>> = Vec::new();
static mut VIS: Vec<bool> = Vec::new();

fn dfs(c: usize) {
    unsafe {
        VIS[c-1] = true;
        for v in G[c-1].iter() {
            if !VIS[v-1] {
                dfs(*v);
            }
        }
    }    
}


fn main() {
    input! {
        n: usize,
        m: usize,
    }

    unsafe {
        G = vec![vec![]; n];
        VIS = vec![false; n];

        for _ in 0..m {
            input! {
                u: usize,
                v: usize,
            }
            G[u-1].push(v);
            G[v-1].push(u);
        }
        // println!("{:?}", G);

        let mut ans: u32 = 0;
        for i in 1..n+1 {
            if !VIS[i-1] {
                ans += 1;
                dfs(i);
            }
        }
        println!("{}", ans);
    }
}
